ipaserver/install/installutils.py

```diff
@@ -1552,3 +1552,20 @@ def validate_mask():
     if mask & 0b111101101 > 0:
         mask_str = "{:04o}".format(mask)
     return mask_str
+
+
+def get_replication_plugin_name(dirsrv_get_entry):
+    # Support renaming of a replication plugin in 389-ds
+    # IPA topology plugin depends on the replication plugin but
+    # 389-ds cannot handle older alias querying in the plugin
+    # configuration with 'nsslapd-plugin-depends-on-named: ..' attribute
+    #
+    # dirsrv_get_entry: function (dn, attrs) -> str
+    # returns entry dictionary
+    try:
+        entry = dirsrv_get_entry(
+            'cn=Multisupplier Replication Plugin,cn=plugins,cn=config',
+            ['cn'])
+        return str(entry['cn'], encoding='utf-8')
```

Comment:
When called from the installer, entry['cn'] is `b'Multisupplier Replication Plugin'` but when called from the upgrade we get a list of str `['Multisupplier Replication Plugin']`.
The methods used as dirsrv_get_entry are so different that it's difficult to factor the code for both cases.